### DISCUSSION TITLE
Github integration in subfolders?

### DISCUSSION THREAD

**Discussion by ericthomas1** [dendronhq/dendron#706] [Q&A] [answered]
Hello,

I'd like to use Dendron **for everything** from daily journaling to code development. Its a bit murky to me on how to integrate Github to various subfolders. 

**Example:**
- I like developing Azure Functions
- They have a [required folder structure](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-python#folder-structure)
- I'd like to have these projects as part of my Dendron workspace
- These projects are under Github version control and will have corresponding `.git` directories, etc. 

Its unclear to me how to keep my Dendron "world" under version control itself as well as projects (subfolders) within it. 

Is anyone doing this yet? Is Dendron able to accommodate this in a way thats not too complex? 

**The issue I'm trying to overcome**
- Whenever I want to work on a project that is under version control, I need to `File/Open Folder` and go to a folder outside my Dendron Workspace. 
- Do work (some of which is documentation for a given project)
- Then `File/Open Workspace` and come back to Dendron to check off To-do's, etc.

**Comment by kpathakota** [comment, marked as answer]
I *think* there's a version of this that could work now where you have a top level git repo mapped to your dendron workspace and related files. However, you've got a bunch of sub-folders with projects. These *must* be in your gitignore otherwise vscode will not pick them up as separate git repositories. 

However, I think to get the full scope of what you're doing, you'd need #322 implemented because then you could have dendron functionality available in your project code. 

Moved this to a discussion so other folks can add their thoughts/ideas.

**Reply by ericthomas1** [reply]
Thank you. #322 looks like the solution. Hope this is implemented! Would make Dendron part of a more cohesive VS Code developer workflow.
